Modules/System/Azure File Services API/src/Helper/AFSWebRequestHelper.Codeunit.al

```diff
@@ -0,0 +1,182 @@
+    var
+        AFSOperationResponse: Codeunit "AFS Operation Response";
+        HttpClient: HttpClient;
+        HttpRequestMessage: HttpRequestMessage;
+    begin
+        PrepareRequestMsg(HttpRequestMessage, AFSOperationPayload, Enum::"Http Request Type"::POST, HttpContent);
+
+        AFSOperationResponse := SendRequest(HttpClient, HttpRequestMessage, OperationNotSuccessfulErr);
+        exit(AFSOperationResponse);
+    end;
+
+    [NonDebuggable]
+    procedure OptionsOperation(var AFSOperationPayload: Codeunit "AFS Operation Payload"; OperationNotSuccessfulErr: Text): Codeunit "AFS Operation Response"
+    var
+        AFSOperationResponse: Codeunit "AFS Operation Response";
+        HttpClient: HttpClient;
+        HttpRequestMessage: HttpRequestMessage;
+    begin
+        PrepareRequestMsg(HttpRequestMessage, AFSOperationPayload, Enum::"Http Request Type"::OPTIONS);
+
+        AFSOperationResponse := SendRequest(HttpClient, HttpRequestMessage, OperationNotSuccessfulErr);
+        exit(AFSOperationResponse);
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(var HttpRequestMessage: HttpRequestMessage; var AFSOperationPayload: Codeunit "AFS Operation Payload"; HttpRequestType: Enum "Http Request Type")
+    var
+        AFSHttpHeaderHelper: Codeunit "AFS HttpHeader Helper";
+        Authorization: Interface "Storage Service Authorization";
+    begin
+        HttpRequestMessage.Method(Format(HttpRequestType));
+        HttpRequestMessage.SetRequestUri(AFSOperationPayload.ConstructUri());
+        AFSHttpHeaderHelper.HandleRequestHeaders(HttpRequestType, HttpRequestMessage, AFSOperationPayload);
+
+        Authorization := AFSOperationPayload.GetAuthorization();
+        Authorization.Authorize(HttpRequestMessage, AFSOperationPayload.GetStorageAccountName());
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(var HttpRequestMessage: HttpRequestMessage; var AFSOperationPayload: Codeunit "AFS Operation Payload"; HttpRequestType: Enum "Http Request Type"; HttpContent: HttpContent)
+    var
+        AFSHttpContentHelper: Codeunit "AFS HttpContent Helper";
+        AFSHttpHeaderHelper: Codeunit "AFS HttpHeader Helper";
+    begin
+        if AFSHttpContentHelper.ContentSet(HttpContent) or AFSHttpHeaderHelper.HandleContentHeaders(HttpContent, AFSOperationPayload) then
+            HttpRequestMessage.Content := HttpContent;
+
+        PrepareRequestMsg(HttpRequestMessage, AFSOperationPayload, HttpRequestType);
+    end;
+
+    [NonDebuggable]
+    local procedure SendRequest(var HttpClient: HttpClient; HttpRequestMessage: HttpRequestMessage; OperationNotSuccessfulErr: Text): Codeunit "AFS Operation Response"
+    var
+        AFSOperationResponse: Codeunit "AFS Operation Response";
+        HttpResponseMessage: HttpResponseMessage;
+    begin
+        if not HttpClient.Send(HttpRequestMessage, HttpResponseMessage) then
+            Error(OperationNotSuccessfulErr);
```

Comment:
Should we mention this error on the summary of the public methods of the Codeunit AFS Client?